lib/sequelize.js

```diff
@@ -455,122 +455,129 @@ class Sequelize {
    *
    * @see {@link Model#build} for more information about instance option.
    */
+
   query(sql, options) {
-    if (arguments.length > 2) {
-      // TODO: Remove this note in the next major version (4.0)
-      throw new Error('Sequelize.query was refactored to only use the parameters `sql` and `options`. Please read the changelog about BC.');
-    }
+    function processQuery() {
```

Comment:
Just pass an arrow function to `Promise.try()`, see the example in the docs I linked